sdk/src/main/java/com/kin/ecosystem/recovery/BackupAndRestoreImpl.java

```diff
@@ -0,0 +1,127 @@
+package com.kin.ecosystem.recovery;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import com.kin.ecosystem.common.KinCallback;
+import com.kin.ecosystem.common.exception.BlockchainException;
+import com.kin.ecosystem.common.exception.ClientException;
+import com.kin.ecosystem.common.exception.KinEcosystemException;
+import com.kin.ecosystem.core.accountmanager.AccountManager;
+import com.kin.ecosystem.core.bi.EventLogger;
+import com.kin.ecosystem.core.bi.events.BackupWalletCompleted;
+import com.kin.ecosystem.core.bi.events.GeneralEcosystemSdkError;
+import com.kin.ecosystem.core.bi.events.RestoreWalletCompleted;
+import com.kin.ecosystem.core.data.blockchain.BlockchainSource;
+import com.kin.ecosystem.core.data.settings.SettingsDataSource;
+import com.kin.ecosystem.core.util.Validator;
+
+public class BackupAndRestoreImpl implements BackupAndRestore {
+
+	protected final BackupManager backupManager;
+	private final AccountManager accountManager;
+	private final EventLogger eventLogger;
+	private final SettingsDataSource settingsDataSource;
+	private final BlockchainSource blockchainSource;
+
+	public BackupAndRestoreImpl(@NonNull final Activity activity, @NonNull AccountManager accountManager,
+		@NonNull EventLogger eventLogger, @NonNull BlockchainSource blockchainSource, SettingsDataSource settingsDataSource) {
+		this.blockchainSource = blockchainSource;
+		this.backupManager = new BackupManager(activity, blockchainSource.getKeyStoreProvider());
+		this.accountManager = accountManager;
+		this.eventLogger = eventLogger;
+		this.settingsDataSource = settingsDataSource;
+	}
+
+	@Override
+	public void backupFlow() throws ClientException {
+		if (blockchainSource.getKinAccount() != null) {
+			backupManager.backupFlow();
+		} else {
+			throw new ClientException(ClientException.ACCOUNT_NOT_LOGGED_IN, "Account should be logged in before backup.", null);
+		}
+	}
+
+	@Override
+	public void restoreFlow() {
+		backupManager.restoreFlow();
```

Comment:
why don't we check here if we're logged in?

Reply:
you are right not sure why.
we should.